File: api-tester-java/src/main/java/com/revenuecat/api_tester_java/CustomerInfoAPI.java
package com.revenuecat.api_tester_java;

import android.net.Uri;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.EntitlementInfos;
import com.revenuecat.purchases.models.Transaction;

import java.util.Date;
import java.util.List;
import java.util.Set;

@SuppressWarnings({"ConstantConditions", "unused", "SpellCheckingInspection"})
final class CustomerInfoAPI {
    static void check() {
        final CustomerInfo customerInfo = null;

        final EntitlementInfos entitlementInfo = customerInfo.getEntitlements();
        final Set<String> asubs = customerInfo.getActiveSubscriptions();
        final Set<String> skus = customerInfo.getAllPurchasedSkus();
        final Date led = customerInfo.getLatestExpirationDate();

        final List<Transaction> nst = customerInfo.getNonSubscriptionTransactions();
        final Date opd = customerInfo.getOriginalPurchaseDate();
        final Date rd = customerInfo.getRequestDate();
        final Date fs = customerInfo.getFirstSeen();
        final String oaui = customerInfo.getOriginalAppUserId();
        final Uri mu = customerInfo.getManagementURL();

        final Date eds = customerInfo.getExpirationDateForSku("");
        final Date pds = customerInfo.getPurchaseDateForSku("");
        final Date ede = customerInfo.getExpirationDateForEntitlement("");
        final Date pde = customerInfo.getPurchaseDateForEntitlement("");
    }
}

File: api-tester-kotlin/src/main/java/com/revenuecat/apitesterkotlin/CustomerInfoAPI.kt
package com.revenuecat.apitesterkotlin

import android.net.Uri
import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.EntitlementInfos
import com.revenuecat.purchases.models.Transaction
import java.util.Date

@Suppress("unused")
private class CustomerInfoAPI {
    fun check(customerInfo: CustomerInfo) {
        val entitlementInfo: EntitlementInfos = customerInfo.entitlements
        val asubs = customerInfo.activeSubscriptions
        val skus: Set<String> = customerInfo.allPurchasedSkus
        val led: Date? = customerInfo.latestExpirationDate
        val nst: List<Transaction> = customerInfo.nonSubscriptionTransactions
        val opd: Date? = customerInfo.originalPurchaseDate
        val rd: Date = customerInfo.requestDate
        val fs: Date = customerInfo.firstSeen
        val oaui: String = customerInfo.originalAppUserId
        val mu: Uri? = customerInfo.managementURL
        val eds: Date? = customerInfo.getExpirationDateForSku("")
        val pds: Date? = customerInfo.getPurchaseDateForSku("")
        val ede: Date? = customerInfo.getExpirationDateForEntitlement("")
        val pde: Date? = customerInfo.getPurchaseDateForEntitlement("")
    }
}
